### ISSUE TITLE
enum_glob_use uses capitalization instead of checking for the right kind of def

### ISSUE THREAD

**Issue by Manishearth** [rust-lang-nursery/rust-clippy#2397]
`enum_glob_use` lints for `use foo::bar::Enum::*`.

However, the lint is written like this:

```rust
        if let ItemUse(ref path, UseKind::Glob) = item.node {
            // FIXME: ask jseyfried why the qpath.def for `use std::cmp::Ordering::*;`
            // extracted through `ItemUse(ref qpath, UseKind::Glob)` is a `Mod` and not an
            // `Enum`
            // if let Def::Enum(_) = path.def {
            if path.segments
                .last()
                .and_then(|seg| seg.name.as_str().chars().next())
                .map_or(false, char::is_uppercase)
            {
                span_lint(cx, ENUM_GLOB_USE, item.span, "don't use glob imports for enum variants");
            }
        }
```

which means that it really lints for `use foo::bar::AnyCapitalizedThing::*`. The only by-style capitalized thing you can import from is an enum, so this is usually ok. However you can have modules not following style, which can lead to false positives. We should fix this.

As the comment said, there's a bit of a kink in that variant imports use a module def instead of an enum def.
